rir/src/compiler/analysis/dead_store.h

```diff
@@ -177,15 +293,28 @@ class DeadStoreAnalysis {
                     effect.update();
                 }
             } else if (i->readsEnv()) {
-                observeFullEnv(i->env());
+                observeFullEnv(aliases.resolveEnv(i->env()));
             } else if (i->exits() || i->effects.contains(Effect::ExecuteCode)) {
```

Comment:
1. I think there is an existing bug here, because if an instruction has both both ExecuteCode and readsEnv, then probably the ExecuteCode case should apply first, since it is the one with more effects.
2. The `Force` instruction you handle in applyCommon has both readsEnv and ExecuteCode, but since you now handle it with a sub-analysis, we don't need to be pessimistic anymore. It would be nice if you can somehow not doe the observeFullEnv iff the sub analysis analyzes the promise.

Reply:
For 2. I think if a `Force` forces a statically known promise, we can remove the ExecuteCode.

Reply:
oh no. Granted we know what we are executing, still we are executing more code... I think of this flag as sort of a stand-in for any effects the code that gets executed could have...

Reply:
and still it also does not fix the fact that we are still doing the readsEnv case to a Force, that we actually handle in applyCommon instead